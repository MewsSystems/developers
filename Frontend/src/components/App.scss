html,
body {
  margin: 0;
  padding: 0;
}

html {
  background: #000;
  color: #fff;
}

// react-paginate library doesn't provide api to replace its componets
// so I didn't find a way to use styled components with it
.Search-Page-Pagination {
  display: flex;
  list-style: none;
  justify-content: center;
  margin: 2rem 0;

  .previous,
  .next {
    display: none;
  }

  li {
    border: 1px solid #fff;
    margin: 0 0.25rem;
    padding: 0;
    cursor: pointer;

    &.break {
      border: none;
    }

    &.selected {
      background: #fff;
      color: #000;
      border-color: #000;
    }
  }

  a {
    display: block;
    padding: 0.5rem;
  }
}
