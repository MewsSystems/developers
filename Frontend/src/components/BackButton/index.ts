export { default } from './BackButton';
export * from './styled';
