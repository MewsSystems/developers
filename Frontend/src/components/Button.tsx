import React from 'react'
import styled, { css } from 'styled-components'
import { BORDER_RADIUS, BOX_SHADOW } from 'constants/index'
import { COLORS } from 'constants/colors'
import { darken } from 'polished'

export const StyledButton = styled.button<
  Pick<ButtonProps, 'disabled' | 'variant' | 'fullWidth'>
>`
  width: ${({ fullWidth }) => fullWidth && '100%'};
  min-width: 12rem;
  padding: 1rem;
  text-align: center;
  text-transform: uppercase;
  border: 0;
  border-radius: ${BORDER_RADIUS.MEDIUM};
  box-shadow: ${BOX_SHADOW.MEDIUM};

  ${({ disabled, variant }) =>
    disabled
      ? css`
          background: ${COLORS.GRAY};
          color: ${COLORS.DARK_GRAY};
        `
      : css`
          background: ${variant === 'primary' ? COLORS.BLUE : COLORS.RED};
          color: ${COLORS.WHITE};
          cursor: pointer;
          transition: background ease-in 0.5s;

          :hover {
            background: ${variant === 'primary'
              ? darken(0.1, COLORS.BLUE)
              : darken(0.1, COLORS.RED)};
            transition: background ease-in 0.5s;
          }
        `}
`

export interface ButtonProps {
  variant?: 'primary' | 'secondary'
  disabled?: boolean
  fullWidth?: boolean
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void
  children?: React.ReactNode | React.ReactNodeArray
}

export const Button: React.FC<ButtonProps> = ({
  variant = 'primary',
  disabled,
  fullWidth,
  onClick,
  children,
}) => (
  <StyledButton
    variant={variant}
    disabled={disabled}
    fullWidth={fullWidth}
    onClick={onClick}
  >
    {children}
  </StyledButton>
)
