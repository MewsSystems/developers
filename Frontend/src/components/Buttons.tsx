import styled from '../utils/styled'

const ThemeSwitcherButton = styled('button')`
  display: inline-block;
  padding: 0.25rem 0.5rem;
  border: 1px solid ${props => props.theme.colors.white};
  border-radius: 3px;
  background-color: ${props => props.theme.colors.white};
  color: ${props => props.theme.colors.brand};
  font-size: 0.8rem;
  text-transform: uppercase;
  letter-spacing: 1px;
  cursor: pointer;
  transition: all 0.3s ease;

  &:hover,
  &:focus {
    background-color: transparent;
    color: ${props => props.theme.colors.white};
  }
`

export default ThemeSwitcherButton
