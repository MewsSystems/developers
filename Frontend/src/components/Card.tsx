import React from 'react'
import styled from 'styled-components'
import truncate from 'truncate'
import { COLORS } from 'constants/colors'
import { BORDER_RADIUS, BOX_SHADOW } from 'constants/index'

export const StyledCard = styled.div<{ background: string }>`
  display: flex;
  flex-direction: column;
  flex-grow: 0;
  justify-content: flex-end;
  height: 18rem;
  padding: 1rem;
  background: url(${({ background }) => background}) no-repeat 50% 50%;
  background-size: cover;
  border: 1px solid ${COLORS.GRAY};
  border-radius: ${BORDER_RADIUS.MEDIUM};
  box-shadow: ${BOX_SHADOW.MEDIUM};
  cursor: pointer;
`

const Title = styled.h2`
  color: ${COLORS.WHITE};
  font-size: 1.2rem;
  text-shadow: 1px 1px 0 ${COLORS.BLACK};
`

const TitleContainer = styled.div`
  display: flex;
  flex-direction: row;
  margin: 0 0 0.3rem;

  ${Title} {
    margin-right: 0.8rem;
  }
`

const Overview = styled.p`
  color: ${COLORS.WHITE};
  font-size: 0.8rem;
  text-shadow: 1px 1px 0 ${COLORS.BLACK};
`

const Language = styled.span`
  align-self: center;
  padding: 0.1rem 0.3rem 0.15rem;
  color: ${COLORS.WHITE};
  font-weight: bold;
  font-size: 0.8rem;
  text-transform: uppercase;
  background: ${COLORS.DARK_GRAY};
  border-radius: ${BORDER_RADIUS.MEDIUM};
`

export interface CardProps {
  id: number
  background: string
  title: string
  overview: string
  overviewMaxLength?: number
  language: string
  onClick?: (id: number) => void
}

export const Card: React.FC<CardProps> = ({
  id,
  background,
  title,
  overview,
  overviewMaxLength = 120,
  language,
  onClick,
}) => (
  <StyledCard background={background} onClick={() => onClick && onClick(id)}>
    <TitleContainer>
      <Title>{title}</Title>
      <Language>{language}</Language>
    </TitleContainer>

    <Overview>{truncate(overview, overviewMaxLength)}</Overview>
  </StyledCard>
)
