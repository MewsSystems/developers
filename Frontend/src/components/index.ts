import styled from "styled-components";
import { Link } from "react-router-dom";

export const Layout = styled.div`
  display: grid;
  grid-template-columns: 350px 1fr 350px;
  height: 100vh;
`;

export const CenterColumn = styled.div`
  grid-column: 2;
  padding: 20px;
`;

export const Card = styled.div`
  box-shadow: 0 8px 16px 0 rgba(0,0,0,0.2);
  display: flex;
  padding: 10px;
  color: gray;
  font: 16px/1.5 Helvetica, Verdana, sans-serif;
`

export const Content = styled.div`
  padding: 2px 16px;
`

export const Title = styled.div`
  display: block;
  margin: 10px 0;
  font-weight: bold;
  color: black;
  font-size:20px;
`;

export const SubTitle = styled.div`
  font-size:14px;
  color:grey;
  margin:2px 0;
`;

export const Paragraph = styled.p`
  line-height: 18px;
  margin: 5px;
`;

export const ErrorMessage = styled.div`
  border:1px solid red;
  padding:10px;
  color:red;
  font: 18px Helvetica, Verdana, sans-serif;
`



export const StyledInput = styled.input`
  background-color: transparent;
  border: none;
  border-bottom: 1px solid #9e9e9e;
  border-radius: 0;
  outline: none;
  height: 3rem;
  width: 100%;
  font-size: 16px;
  margin: 0 0 8px 0;
  padding: 0;
  box-shadow: none;
  box-sizing: content-box;
  transition: box-shadow .3s, border .3s;
`;

export const StyledLabel = styled.label`
  width:100%;
  color: #9e9e9e;
  position: absolute;
  top: -15px;
  left: 0;
  font-size: 1rem;
  cursor: text;
  transition: transform .2s ease-out, color .2s ease-out;
  text-align: initial;
  &::after {
    display: block;
    content: "";
    position: absolute;
    top: 100%;
    left: 0;
    opacity: 0;
    transition: .2s opacity ease-out, .2s color ease-out;
  }
`

export const Button = styled.button`
  text-decoration: none;
  color: #fff;
  background-color: #26a69a;
  text-align: center;
  letter-spacing: .5px;
  transition: background-color .2s ease-out;
  cursor: pointer;
  font-size: 14px;
  outline: 0;
  border: none;
  border-radius: 2px;
  display: inline-block;
  height: 36px;
  line-height: 36px;
  padding: 0 16px;
  text-transform: uppercase;
  vertical-align: middle;
  &:disabled {
    background-color:#ccc;
  }
`;

export const StyledLink = styled(Link)`
  text-decoration: none;
  color: #26a69a;
`;
