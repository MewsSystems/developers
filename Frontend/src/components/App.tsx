import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Search from './Search/Search';
import Movie from './Movie/Movie';
import { Provider } from 'react-redux';
import createStore from '../store';
import './App.scss';

const store = createStore();

const App = () => {
  return (
    <Provider store={store}>
      <Router>
        <Switch>
          <Route path="/" exact component={Search} />
          <Route path="/movie/:id" render={({ match }) => <Movie id={match.params.id} />} />
        </Switch>
      </Router>
    </Provider>
  );
};

export default App;
