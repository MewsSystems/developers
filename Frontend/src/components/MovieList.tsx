import React from 'react'
import { MovieCard } from './MovieCard'
import styled from 'styled-components'
import { Movie } from 'model/api/Movie'
import { Button } from './Button'
import { Spacing } from './Spacing'
import { Card as LoadingCard } from 'components/Loading/Card'
import { ConfigurationBackdropSizesEnum } from 'model/api/ConfigurationBackdropSizesEnum'
import { useTranslation } from 'react-i18next'
import { useGetImagePath } from 'hooks/useGetImagePath'

const Grid = styled.div`
  display: grid;
  grid-gap: 1rem;
  grid-template-columns: 1fr 1fr 1fr;
`

export interface MovieListProps {
  items: Movie[]
  page: number
  totalPages: number
  loading: boolean
  onLoadMore: () => void
  onMovieClick?: (id: number) => void
}

export const MovieList: React.FC<MovieListProps> = ({
  items,
  page,
  totalPages,
  loading,
  onLoadMore,
  onMovieClick,
}) => {
  const { t } = useTranslation()
  const getImagePath = useGetImagePath(ConfigurationBackdropSizesEnum.W_780)

  if (items.length > 0) {
    return (
      <>
        <Grid>
          {items.map(
            ({ title, overview, backdrop_path, original_language, id }) => (
              <MovieCard
                key={id}
                id={id}
                title={title}
                overview={overview}
                background={getImagePath(backdrop_path)}
                language={original_language}
                onClick={onMovieClick}
              />
            )
          )}
          {loading && (
            <>
              <LoadingCard />
              <LoadingCard />
              <LoadingCard />
              <LoadingCard />
              <LoadingCard />
            </>
          )}
        </Grid>
        <Spacing outer="2rem 0 0">
          <Button
            onClick={onLoadMore}
            disabled={page === totalPages || loading}
            fullWidth
          >
            {t('load-more')}
          </Button>
        </Spacing>
      </>
    )
  }

  return null
}
