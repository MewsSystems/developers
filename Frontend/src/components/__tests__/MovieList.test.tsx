import { MovieList } from 'components/MovieList'
import 'jest-styled-components'
import React from 'react'
import { movieMock } from 'test/data/movies'
import { shallowWithStore } from 'test/shallowWithStore'

describe('Test MovieList component', () => {
  it('Tests snapshot with no items', () => {
    expect(
      shallowWithStore(
        <MovieList
          items={[]}
          loading={false}
          onLoadMore={jest.fn()}
          onMovieClick={jest.fn()}
          page={0}
          totalPages={0}
        />
      )
    ).toMatchSnapshot()
  })

  it('Tests snapshot with items', () => {
    expect(
      shallowWithStore(
        <MovieList
          items={[movieMock]}
          loading={false}
          onLoadMore={jest.fn()}
          onMovieClick={jest.fn()}
          page={1}
          totalPages={1}
        />
      )
    ).toMatchSnapshot()
  })
})
