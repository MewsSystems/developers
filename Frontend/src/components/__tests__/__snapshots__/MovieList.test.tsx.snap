// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MovieList component Tests snapshot with items 1`] = `
<Fragment>
  <styled.div>
    <MovieCard
      background="testw780test.jpg"
      id={1}
      key="1"
      language="en"
      onClick={[MockFunction]}
      overview="test test"
      title="test"
    />
  </styled.div>
  <Spacing
    outer="2rem 0 0"
  >
    <Button
      disabled={true}
      fullWidth={true}
      onClick={[MockFunction]}
    >
      load-more
    </Button>
  </Spacing>
</Fragment>
`;

exports[`Test MovieList component Tests snapshot with no items 1`] = `""`;
